File: src/main/java/NekoEssentials/commands/rename.java
package NekoEssentials.commands;

import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;

import static NekoEssentials.Main.c;
import static NekoEssentials.Main.config;

public class rename implements CommandExecutor {

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        if (label.equalsIgnoreCase("rename")) {
            if (sender instanceof Player) {
                Player player = (Player) sender;
                if (player.hasPermission("NekoEssentials.rename")) {
                    if (args.length >= 1) {
                        ItemStack itemInHand = player.getInventory().getItemInHand();
                        ItemMeta itemMeta = itemInHand.getItemMeta();

                        StringBuilder newName = new StringBuilder();
                        for (String word : args) {
                            newName.append(word).append(" ");
                        }

                        newName = new StringBuilder(ChatColor.translateAlternateColorCodes('&', newName.toString().trim()));

                        itemMeta.setDisplayName(newName.toString());
                        itemInHand.setItemMeta(itemMeta);

                        player.sendMessage(c(config.getConfig().getString("rename-success")));
                    } else {
                        player.sendMessage(c(config.getConfig().getString("rename-usage")));
                    }
                } else {
                    player.sendMessage(c(config.getConfig().getString("no-permission")));
                }
            }
        }
        return false;
    }
}

File: src/main/java/NekoUtility/commands/rename.kt
package NekoUtility.commands

import NekoUtility.Main
import NekoUtility.Main.Companion.c
import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class rename : CommandExecutor {
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        if (sender is Player) {
            val player = sender
            if (player.hasPermission("nekoutility.rename")) {
                if (args.size >= 1) {
                    val itemInHand = player.inventory.itemInHand
                    val itemMeta = itemInHand.itemMeta
                    var newName = StringBuilder()
                    for (word in args) {
                        newName.append(word).append(" ")
                    }
                    newName = StringBuilder(ChatColor.translateAlternateColorCodes('&', newName.toString().trim { it <= ' ' }))
                    itemMeta.displayName = newName.toString()
                    itemInHand.setItemMeta(itemMeta)
                    player.sendMessage(c(Main.config!!.getConfig().getString("rename-success")))
                } else {
                    player.sendMessage(c(Main.config!!.getConfig().getString("rename-usage")))
                }
            } else {
                player.sendMessage(c(Main.config!!.getConfig().getString("no-permission")))
            }
        }

        return false
    }
}